Traceback (most recent call last):
  File "/Users/joelchan/opt/anaconda3/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/Users/joelchan/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 1112, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/joelchan/opt/anaconda3/lib/python3.9/site-packages/nbclient/util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/joelchan/opt/anaconda3/lib/python3.9/site-packages/nbclient/util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "/Users/joelchan/opt/anaconda3/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/Users/joelchan/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 553, in async_execute
    await self.async_execute_cell(
  File "/Users/joelchan/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 857, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/Users/joelchan/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 760, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
path = "assets/"
fname = "newfile.txt"
fpath = f"{path}{fname}"

# basic write
# put 'w' as the seocnd argument
fhand = open(fpath, 'w')
print(f)
fhand.write("Hello world!")
fhand.close()
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
[0;32m/var/folders/xz/_hjc5hsx743dclmg8n5678nc0000gn/T/ipykernel_11088/2055429998.py[0m in [0;36m<module>[0;34m[0m
[1;32m      6[0m [0;31m# put 'w' as the seocnd argument[0m[0;34m[0m[0;34m[0m[0m
[1;32m      7[0m [0mfhand[0m [0;34m=[0m [0mopen[0m[0;34m([0m[0mfpath[0m[0;34m,[0m [0;34m'w'[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m----> 8[0;31m [0mprint[0m[0;34m([0m[0mf[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m      9[0m [0mfhand[0m[0;34m.[0m[0mwrite[0m[0;34m([0m[0;34m"Hello world!"[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m     10[0m [0mfhand[0m[0;34m.[0m[0mclose[0m[0;34m([0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;31mNameError[0m: name 'f' is not defined
NameError: name 'f' is not defined

